roles/addons/diskless/files/livenet_module.py

```diff
@@ -221,8 +221,8 @@ def generate_squashfs(self):
         os.system('mount -o loop ' + self.IMAGE_DIRECTORY + 'tosquash/LiveOS/rootfs.img ' + self.MOUNT_DIRECTORY)
 
         # Put the operating system inside rootfs.img
-        logging.debug('Executing \'cp -r ' + self.WORKING_DIRECTORY + 'generated_os/* ' + self.MOUNT_DIRECTORY + '\'')
-        os.system('cp -r ' + self.WORKING_DIRECTORY + 'generated_os/* ' + self.MOUNT_DIRECTORY)
+        logging.debug('Executing \'cp -a ' + self.WORKING_DIRECTORY + 'generated_os/* ' + self.MOUNT_DIRECTORY + '\'')
+        os.system('cp -a ' + self.WORKING_DIRECTORY + 'generated_os/* ' + self.MOUNT_DIRECTORY)
```

Comment:
Why use cp instead of rsync?
I would tend to use Python's `shutil.copytree()` instead of `os.system()`.

Reply:
Hello, this are good alternatives but I am not sure this is the right place for the change? NFS images also use `cp -a`.
Also I am not sure how to best use `shutil.copytree()` in this situation, it is known to cause problems when the destination directory already exists. For example I tried this:
```
shutil.copytree(self.WORKING_DIRECTORY + 'generated_os/', self.MOUNT_DIRECTORY)
```
And I get this error:
```
FileExistsError: [Errno 17] File exists: '/var/tmp/diskless/workdir/kitchen_image/mnt/'
```

Reply:
The directory "self.WORKING_DIRECTORY + 'generated_os/" is not supposed to be already filled at this time, it is not a problem to use shutils.copytree() here.
I think use shutil.copytree() is recommanded here because it use directly Python commands instead of Linux system commands.

Reply:
`dirs_exist_ok=True` should not raise an exception in your case.
Anyway, it is indeed not the right place to change this. I was just wondering why the use of `os.system()` instead of Python's built-in methods.